cpp/log/sqlite_db-inl.h

```diff
@@ -317,9 +328,27 @@ template <class Logged>
 int64_t SQLiteDB<Logged>::TreeSize() const {
   cert_trans::ScopedLatency latency(
       latency_by_op_ms.GetScopedLatency("tree_size"));
-  std::lock_guard<std::mutex> lock(lock_);
+  std::unique_lock<std::mutex> lock(lock_);
 
   CHECK_GE(tree_size_, 0);
+  sqlite::Statement statement(
```

Comment:
I just send #731 to cache the staleness status in `HttpHandler`. Once that's merged in, I'd say that declaring `Database<>::TreeSize()` to be blocking would be okay.